### PR TITLE
Docs: Doc of tracking add2cart event

### PR DESCRIPTION
**Pull request by mariorey**
Added documentation about how to track add2cart event
# Pull request template
 
Describe the purpose of the change, the specific changes done in detail, and the issue you have fixed.

## Motivation and context

- [ ] Dependencies. If any, specify:
- [X] Open issue. If applicable, link: https://searchbroker.atlassian.net/browse/EMP-2699

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [X] `Main`
- [ ] Other. Specify:

## How has this been tested?

Tests performed according to [testing guidelines](./contributing/tests.md): 

## Checklist:

- [ ] My code follows the **[style guidelines](./CONTRIBUTING.md#style-guides)** of this project.
- [ ] I have performed a **self-review** on my own code.
- [ ] I have **commented** my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the **[documentation](./CONTRIBUTING.md#documentation-style-guide)**.
- [ ] My changes generate **no new warnings**.
- [ ] I have added **tests** that prove my fix is effective or that my feature works.
- [ ] New and existing **unit tests pass locally** with my changes.

### CODE REVIEW

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md [hunk 1]
```diff
@@ -405,3 +405,60 @@ instance:
...
+```html
+window.InterfaceX.init({
+    instance: "instanceName",
+    lang: "es",
+    documentDirection: "ltr",
+    scope: t,
+    currency: "EUR",
+    consent: !0,
+    env: void 0,
+    isSPA: !0,
```
Comment (herrardo) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:425]: My example in the task was wrong, I don't know where I copied those values but are not descriptive for developers that have to integrate this:

```suggestion
    scope: 'desktop',
    currency: "EUR",
    consent: true,
    isSPA: false,
```

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md [hunk 2]
```diff
@@ -405,3 +405,60 @@ instance:
   });
 </script>
 ```
+
+#### Tracking the Add2Cart event from PDP
```
Comment (herrardo) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:409]:
```suggestion
### Tracking the Add2Cart event from PDP
```

This is the correct heading for a new section to be in the same level as the other `Notes on X Archetype integration`

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md [hunk 3]
```diff
@@ -405,3 +405,60 @@ instance:
...
+
+The X interface incorporates a mechanism that enables the tracking of the client's Add2Cart interaction by the shopper. The approach to achieving this varies depending on the website's structure.
+
+**NON-SPA**
```
Comment (herrardo) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:413]:
```suggestion
#### NON-SPA
```

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md [hunk 4]
```diff
@@ -405,3 +405,60 @@ instance:
...
+
+2. Once X is initialized the `InterfaceX.addProductToCart('11776347-ES');` function should be called when the add to cart button is clicked.
+
+```html
+addToCartButton.addEventListener('click', () =>
+  InterfaceX.addProductToCart('11776347-ES');
+);
+```
+
+**SPA**
```
Comment (herrardo) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:447]:
```suggestion
#### SPA
```

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md [hunk 5]
```diff
@@ -405,3 +405,60 @@ instance:
...
+if (yourCommerceStoreEnvironment.isPDP && InterfaceX.getSnippetConfig.isSpa) {
+  InterfaceX.bus.emit('PDPIsLoaded')
+}
+```
+
+2. When the add to cart button is clicked :  
+`InterfaceX.addProductToCart()`
+
+```html
+addToCartButton.addEventListener('click', () =>
```
Comment (herrardo) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:461]:
```suggestion
yourCommerceStoreEnvironment .addToCartButton.addEventListener('click', () =>
```

In this way they don't think that addToCartButton is a button from Empathy env

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md [hunk 6]
```diff
@@ -405,3 +405,60 @@ instance:
...
+        UserClickedAResult: function(a, b, e, t) {}
+    },
+    productId: '11776347-ES' // Add this parameter
+})
+```
+
+2. Once X is initialized the `InterfaceX.addProductToCart('11776347-ES');` function should be called when the add to cart button is clicked.
+
+```html
+addToCartButton.addEventListener('click', () =>
```
Comment (herrardo) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:442]:
```suggestion
yourCommerceStoreEnvironment .addToCartButton.addEventListener('click', () =>
```

In this way they don't think that addToCartButton is a button from Empathy env

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md [hunk 7]
```diff
@@ -405,3 +405,60 @@ instance:
...
+  InterfaceX.addProductToCart('11776347-ES');
+);
+```
+
+**SPA**
+1. When a new page that is of type "Product Details Page" is loaded, this function should be called:
+`InterfaceX.bus.emit('PDPIsLoaded')`
+
+```html
+if (yourCommerceStoreEnvironment.isPDP && InterfaceX.getSnippetConfig.isSpa) {
```
Comment (herrardo) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:452]:
```suggestion
if (yourCommerceStoreEnvironment.isPDP && window.initX.isSpa) {
```

I know I told you to place something as getSnippetConfig, but it doesn't exist, so we can use the `initX` object

**Comment by herrardo** [comment]
Do not merge, waiting for @mlcorton review. :)

**Comment by mlcorton** [comment]
Hi there @mariorey! I've noticed that code snippets are breaking numbered lists. I've added a new commit to fix it (you need to indent code twice to solve it). Also, I've rewritten some content to make it clearer and match the current style and sentence structure. Tell me if I've misled any concept. :) 
cc: @herrardo

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md [hunk 8]
```diff
@@ -406,59 +407,63 @@ instance:
...
+        ],
+        callbacks: {
+            UserClickedAResult: function(a, b, e, t) {}
+        },
+        productId: '11776347-ES' // Add this parameter
+    })
+    ```
 
-The X interface incorporates a mechanism that enables the tracking of the client's Add2Cart interaction by the shopper. The approach to achieving this varies depending on the website's structure.
+2. Call the `InterfaceX.addProductToCart('11776347-ES')` function to track the event when the add-to-cart button is clicekd.
```
Comment (herrardo) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:441]:
```suggestion
2. Call the `InterfaceX.addProductToCart('11776347-ES')` function to track the event when the add-to-cart button is clicked.
```

**Review by herrardo — CHANGES_REQUESTED** [review]

**Review by mlcorton — APPROVED** [review]

**Review by herrardo — APPROVED** [review]

**Pull request merged**